### PR TITLE
chore(customresource): add ArtifactPath method

### PR DESCRIPTION
**Pull request by efekarakus**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/deploy/upload/customresource/customresource.go [hunk 1]
```diff
@@ -54,31 +54,44 @@ var (
...
 
-// Zip creates a zip archive from all the files in the custom resource and returns a reader for its content.
-func (cr *CustomResource) Zip() (io.Reader, error) {
+// ArtifactPath returns the S3 key for the zipped custom resource object.
+func (cr *CustomResource) ArtifactPath() string {
+	return artifactpath.CustomResource(strings.ToLower(cr.FunctionName()), cr.zip.Bytes())
+}
+
+// Zip returns a reader for the zip archive from all the files in the custom resource.
+func (cr *CustomResource) Zip() io.Reader {
```
Comment (Lou1415926) [inline internal/pkg/deploy/upload/customresource/customresource.go]: Do these methods need to be exported ?

File: internal/pkg/aws/s3/s3.go [hunk 2]
```diff
@@ -155,6 +155,11 @@ func ParseURL(url string) (bucket string, key string, err error) {
 	return
 }
 
+// URL returns a virtual-hosted–style S3 url for the object stored at key in a bucket created in the specified region.
+func URL(region, bucket, key string) string {
```
Comment (Lou1415926) [inline internal/pkg/aws/s3/s3.go:161]: I guess this function is getting referenced in an upcoming PR is that correct 👀 ?

**Review by Lou1415926 — APPROVED** [review]

File: internal/pkg/aws/s3/s3.go [hunk 3]
```diff
@@ -155,6 +155,11 @@ func ParseURL(url string) (bucket string, key string, err error) {
...
+	return fmt.Sprintf("https://%s.s3.%s.amazonaws.com/%s", bucket, region, key)
```
Comment (iamhopaul123) [inline internal/pkg/aws/s3/s3.go]: potential partition problem here? maybe not a big deal because we don't have any customers there

**Review by iamhopaul123 — APPROVED** [review]

File: internal/pkg/deploy/upload/customresource/customresource.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/deploy/upload/customresource/customresource.go]: Done!

File: internal/pkg/aws/s3/s3.go (hunk 3 above)
Comment (efekarakus) [reply internal/pkg/aws/s3/s3.go]: Done! I only did this for the CN partition because I couldn't figure out if for other paritions the URL changes X_X

**Pull request merged**